### PR TITLE
feat(z-image): add Z-Image base model integration and trainside recipe

### PR DESCRIPTION
**Pull request by xshrz**
## Summary

Add Z-Image text-to-image model support to UniRL, following the existing
qwen_image / sd3 model-bundle conventions. This introduces a new
`unirl/models/z_image` bundle (config, bundle loader, conditions, text
embedding, VAE decode, diffusion stage, pipeline) and a FlowGRPO trainside
recipe. Defaults target the **base** Z-Image model
(`Tongyi-MAI/Z-Image`): static `shift=6.0`, `num_inference_steps=16`,
`guidance_scale=4.0` (CFG enabled). The text encoder uses the Qwen3 chat
template and the second-to-last hidden state; the FlowMatch schedule uses a
static shift to match the model's `scheduler_config.json`
(`use_dynamic_shifting=false`).

## Related Issue

N/A

## Test Plan

- Hydra config validation:
  `python -m unirl.train_diffusion --config-name=diffusion/z_image_trainside --cfg job --resolve`
- 2-GPU H20 trainside smoke run (overrides: `+devices_per_node=2`) — completed
  successfully, rollout + optimizer step run end to end.
- 4-node / 32-GPU full training run with wandb — pending (not yet run).

## Compatibility / Risk

New, self-contained model bundle and recipe; no changes to existing models or
shared code paths. Requires downloading the `Tongyi-MAI/Z-Image` checkpoint.
To run the Turbo variant instead, override
`pretrained_model_ckpt_path=Tongyi-MAI/Z-Image-Turbo`, `shift=3.0`,
`num_inference_steps=8`, `guidance_scale=0.0`.

## Reviewer Notes

Opened as a draft for early feedback while the full 4x8 multinode run is in
progress. AI assistance was used to scaffold the bundle from the qwen_image /
sd3 references. Please sanity-check the timestep convention (`1 - sigma`),
velocity negation, and CFG formula in `diffusion.py` first.

## Checklist

- [x] I reviewed the changed code and removed unrelated/generated artifacts.
- [x] I updated tests, docs, and configs where needed, or explained why not.

### CODE REVIEW

**Comment by celve** [comment]
We need a full run with reward curve.

**Comment by xshrz** [comment]
ca346c376f6ffc8ef1fe1f7b9ad0247b
bash examples/run_experiment_multinode_taiji.sh diffusion/z_image_trainside

**Review by haonan3 — APPROVED** [review]
LGTM
